esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2298,6 +2264,66 @@ bool CFileSprayEx::onSprayVariable(IEspContext &context, IEspSprayVariable &req,
     return true;
 }
 
+void CFileSprayEx::checkDZScopeAccessAndSetSpraySourceDFUFileSpec(IEspContext &context, const char *srcPlane, const char *srcHost,
+    const char *srcFile, MemoryBuffer &srcXML, IDFUfileSpec *srcDFUfileSpec)
+{
+    if(srcXML.length() == 0)
+    {
+        RemoteMultiFilename rmfn;
+        SocketEndpoint ep(srcHost);
+        if (ep.isNull())
+            throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Cannot resolve source network IP from %s.", srcHost);
+
+        rmfn.setEp(ep);
+        StringBuffer fnamebuf(srcFile);
+        fnamebuf.trim();
+        rmfn.append(fnamebuf.str());    // handles comma separated files
+        srcDFUfileSpec->setMultiFilename(rmfn);
+
+        ForEachItemIn(i, rmfn)
+        {
+            StringBuffer path;
+            rmfn.item(i).getLocalPath(path);
+
+            //A spray source could be: '/path/*'.
+            StringBuffer dirPath, dirTail;
+            splitFilename(path, &dirPath, &dirPath, &dirTail, &dirTail);
+
+            const char* s = streq(dirTail.str(), "*") ? dirPath : path;
```

Comment:
What about e.g.: "/var/lib/HPCCSystmes/mydropzone/dir1/dir2/filex*"
Or even : "/var/lib/HPCCSystmes/mydropzone/dir*blah/fixed"

Also, as it stands, if path was e.g. "/var/lib/HPCCSystmes/mydropzone/dir1/dir2/*",
it strips the wildcard, and scope checks "/var/lib/HPCCSystmes/mydropzone/dir1/dir2",
but if it was /var/lib/HPCCSystmes/mydropzone/dir1/dir2/afile, it would pass the whole thing, meaning it would also scope check "afile" ?

Something doesn't seem quite right here.

Reply:
In the existing code, if the path is "/var/lib/HPCCSystmes/mydropzone/f1/f1*1/file", I got the error from dfuserver:  "Could not open source file /var/lib/HPCCSystmes/mydropzone/f1/f1*1/file". Not sure that format should be supported in ws_fs.

Reply:
I don't know what it supports. Where did the concept of only supporting a trailing wildcard come from?
If we're not sure, and if wildcards in directories are/have never been supported, still need to check that wildcards within (or at the head) of the filename portion work.
(should also add a comment here to state findings that dfuserver only supports X)

Reply:
re. point 2, differential numbers of scopes passed depending on whether a filename without a wildcard (whole thing passed), or a filename with, passes dir part only.

I think all the calls to getDropZoneScopePermissions need checking carefully, I have a feeling some are passing the scoped directory (without trailing path) and some are passing a scoped path including the file, which will lead to inconsistent scopes being checked in LDAP.